features/facebookImport/src/model/entities/facebook-processed-data.js

```diff
@@ -0,0 +1,65 @@
+import { ProcessedData } from "@polypoly-eu/poly-import";
+
+export default class FacebookProcessedData extends ProcessedData {
+    constructor() {
+        super();
+        let self = this;
+        this._picturesCount = null;
```

Comment:
This will not allow to run a loop over results, for instance. Again, I don't think we shoud use attributes for results of analysis; a simple "results" attribute, in the base class, would do. You could easily add a loop over them, or organize them by sections or any way you want. New results can be added easily, too.

Reply:
Well, you could loop over all properties over the `ProcessedData` instance, but I agree, I don't like the design that essentially hard codes all analyses. The analyses where meant to be an _extensible_ system - a `FacebookAccount` does _not_ know what kind of analysis will run on it and what kind of results they will produce. The only reason in my book why we had an `analyses` property on the account object before, was because we're using the account everywhere we use the analyses - but logically it's completely separate to me.

Illustrated another way:

1. Physical files (JSON, HTML etc) -> Parsed by **importers** into
2. Concrete account object (base class being `DataAccount` as of this PR) -> Processed by **analyses**, which produce
3. Analysis results (array of analysis result objects) -> Processed by **stories**, which figure out how to best display these analysis result objects to the user

Reply:
I still don't think we should attempt to hard code potential analysis results and I'm hoping for this code to go away again, but two smaller comments anyway:

1. IMHO, initialising a bunch of properties to `null` doesn't make sense here: It's easy to add some new property from an analysis and forget to set it to `null` here - then if someone attempts to read that, it'd be `undefined`. By not initialising any of these, they would all be `undefined`, which means "not there" -> It's more logical, less boiler plate, and we wouldn't accidentally loop over properties that aren't there.
2. The `_` prefix is customary for private properties, which shouldn't be accessed by anything but the object's own methods. That's not quite the case here :D

But like I said, we should rather go back to the target design.